devnet-sdk/system/wallet.go

```diff
@@ -97,6 +100,86 @@ func (w *wallet) Balance() types.Balance {
 	return types.NewBalance(balance)
 }
 
+func (w *wallet) InitiateMessage(chainID types.ChainID, target common.Address, message []byte) types.WriteInvocation[any] {
+	return &initiateMessageImpl{
+		chain:     w.chain,
+		processor: w,
+		from:      w.address,
+		target:    target,
+		chainID:   chainID,
+		message:   message,
+	}
+}
+
+type initiateMessageImpl struct {
+	chain     internalChain
+	processor TransactionProcessor
+	from      types.Address
+
+	target  types.Address
+	chainID types.ChainID
+	message []byte
+}
+
+func (i *initiateMessageImpl) Call(ctx context.Context) (any, error) {
+	builder := NewTxBuilder(ctx, i.chain)
+	messenger, err := i.chain.ContractsRegistry().L2ToL2CrossDomainMessenger(constants.L2ToL2CrossDomainMessenger)
+	if err != nil {
+		return nil, fmt.Errorf("failed to init transaction: %w", err)
+	}
+	data, err := messenger.ABI().Pack("sendMessage", i.chainID, i.target, i.message)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build calldata: %w", err)
+	}
+	tx, err := builder.BuildTx(
+		WithFrom(i.from),
+		WithTo(constants.L2ToL2CrossDomainMessenger),
+		WithValue(big.NewInt(0)),
+		WithData(data),
+	)
```

Comment:
You might consider using @janjakubnanista's FeeEstimator https://github.com/ethereum-optimism/optimism/pull/14721/files#diff-58e90816b75e34cb65fb1551242155448d0fcc0e54c45ab6ea116a30cb6f4f54R21

Reply:
FeeEstimator may be useful if you directly use the eth client to build the tx data, as shown in jan's test.

Reply:
I re-read https://github.com/ethereum-optimism/optimism/pull/14721's description, and 
> This pricing mechanism can later be included in the transactionProcessor for the bindings covered by devnet-sdk.

so FeeEstimator will be eventually integrated to TxBuilder. cc @janjakubnanista